Spatial.gd

```diff
@@ -339,7 +340,11 @@ func decode_context_packet(spb: StreamPeerBuffer):
 
 	if self.map_id != old_map_id:
 		print("New Map")
-		print("Saving Old Map")
+		if old_map_id != 0:
+			if not read_hash(old_map_id) == get_hash(self.waypoint_data.to_bytes()):
```

Comment:
These two lines might be able to be combined with an `and` or `&&` I forget which syntax gdscript uses, most languages stop evaluating an `and` if the left half is false.